go/cli/mcap/cmd/metadata.go

```diff
@@ -88,6 +98,123 @@ var listMetadataCmd = &cobra.Command{
 	},
 }
 
+var addMetadataCmd = &cobra.Command{
+	Use:   "metadata",
+	Short: "Add metadata to an mcap file",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		if len(args) != 1 {
+			die("Unexpected number of args")
+		}
+		filename := args[0]
+		tempName := filename + ".new"
+		tmpfile, err := os.Create(tempName)
+		if err != nil {
+			die("failed to create temp file: %s", err)
+		}
+		keyvals := make(map[string]string)
+		for _, kv := range addMetadataKeyValues {
+			parts := strings.FieldsFunc(kv, func(c rune) bool {
+				return c == '='
+			})
+			if len(parts) != 2 {
+				die("failed to parse key/value %s", kv)
+			}
+			keyvals[parts[0]] = parts[1]
+		}
+		err = utils.WithReader(ctx, filename, func(remote bool, rs io.ReadSeeker) error {
+			if remote {
+				die("not supported on remote mcap files")
+			}
+			return utils.RewriteMCAP(tmpfile, rs, func(w *mcap.Writer) error {
+				return w.WriteMetadata(&mcap.Metadata{
+					Name:     addMetadataName,
+					Metadata: keyvals,
+				})
+			})
+		})
+		if err != nil {
+			die("failed to add metadata: %s", err)
+		}
+		err = os.Rename(tempName, filename)
+		if err != nil {
+			die("failed to rename temporary output: %s", err)
+		}
+	},
+}
+
+var getMetadataCmd = &cobra.Command{
+	Use:   "metadata",
+	Short: "get metadata by name",
+	Run: func(cmd *cobra.Command, args []string) {
+		ctx := context.Background()
+		if len(args) != 1 {
+			die("Unexpected number of args")
+		}
+		filename := args[0]
+		err := utils.WithReader(ctx, filename, func(_ bool, rs io.ReadSeeker) error {
```

Comment:
in general, the run commands in this patch are too heavy. It would be easier to test if these dispatched to standalone functions.